src/org/openbravo/configuration/ConfigurationApp.java

```diff
@@ -916,14 +916,22 @@ private static void setValuesProperties() {
           replaceProperties.put(PREFIX_DB_PASS, optionLastForReplace.getChosenOption());
         } else if (optionLastForReplace.getAskInfo().equals(DB_SERVER)) {
           serverBBDD = optionLastForReplace.getChosenOption();
-        } else if (optionLastForReplace.getAskInfo().equals(DB_SERVER_PORT)) {
-          portBBDD = optionLastForReplace.getChosenOption();
+        } else if (optionLastForReplace.getAskInfo().equals(PREFIX_DB_URL)) {
+          // Contains the full 'bbdd.url' path
+          urlBBDD = optionLastForReplace.getChosenOption();
         }
       }
+
       replaceProperties.put(PREFIX_DB_RDBMS, "ORACLE");
       replaceProperties.put(PREFIX_DB_DRIVER, "oracle.jdbc.driver.OracleDriver");
-      replaceProperties.put(PREFIX_DB_URL,
-          "jdbc:oracle:thin:@" + serverBBDD + ":" + portBBDD + ":" + nameBBDD);
+
+      if (urlBBDD.equals("")) {
```

Comment:
It not null safe, you may be use StringUtils Java libraries to compare String objects.